lua/lint/linters/inko.lua

```diff
@@ -0,0 +1,70 @@
+local util = require('lint.util')
+
+local severities = {
+  error = vim.lsp.protocol.DiagnosticSeverity.Error,
+  warning = vim.lsp.protocol.DiagnosticSeverity.Warning,
+}
+
+-- Inko unit tests require the inclusion of an extra directory, otherwise we
+-- won't be able to find some of the files imported into unit tests.
+local function include_tests()
+  local path = vim.fn.expand('%:p')
+  local separator = vim.fn.has('win32') == 1 and '\\' or '/'
+  local find = 'tests' .. separator .. 'test' .. separator
+
+  if not path:match(find) then
+    return
+  end
+
+  local tests_dir = util.find_nearest_directory('tests')
+
+  return '--include=' .. tests_dir
+end
+
+return {
+  cmd = 'inko',
+  args = {
+    'build',
+    include_tests,
+    '--format=json',
+    '--check',
+  },
+  stream = 'stderr',
+  stdin = false,
+  ignore_exitcode = true,
+  parser = function(output, _)
+    local items = {}
+
+    if output == '' then
+      return items
+    end
+
+    local decoded = vim.fn.json_decode(output) or {}
+    local bufpath = vim.fn.expand('%:p')
+
+    for _, diag in ipairs(decoded) do
+      if diag.file == bufpath then
```

Comment:
I forgot to add this earlier, but this is needed since diagnostics for other files (e.g. those processed before the one being linted) shouldn't show up for the buffer being linted.